packages/knit-gateway/src/gateway.ts

```diff
@@ -100,6 +111,39 @@ export interface EntryPoint {
   timeoutMs?: number;
 }
 
+/**
+ * Describes a Knit relation.
+ *
+ * @internal
+ */
+export interface Relation {
+  /**
+   * The message type of the base message.
+   */
+  base: MessageType;
+  /**
+   * The relation field info.
+   */
+  field: FieldInfo;
+  /**
+   * The message type of the params.
+   */
+  params?: MessageType;
+  /**
+   * The runtime to use for the relation.
+   *
+   * This is the runtime of the file it was defined in.
+   */
+  runtime: MessageType["runtime"];
+  /**
+   * The resolver to use for the relation.
+   */
+  resolver: (
+    bases: AnyMessage[],
+    params: PartialMessage<AnyMessage> | undefined
```

Comment:
My idea is that this will be done by either the plugin generated code/ more likely the runtime registration API

Reply:
I think maybe we're saying the same thing? If it's done by the runtime registration API, then it could just be `PartialMessage<AnyMessage>` here, right? Because the registration logic would handle creating an empty message when it sees `undefined` (e.g. query that is does not specify any params), so the resolver function never has to worry about receiving and handling an undefined.

Reply:
This is how I intend to use it: https://github.com/bufbuild/knit-ts/pull/9/files#diff-e6f890b397a146d571932f33b43023467b59fd7649c60780efc2750fc841194cR281-R311